src/cargo/ops/registry.rs

```diff
@@ -45,8 +51,13 @@ pub fn publish(manifest_path: &Path,
                                     false, true)).unwrap();
 
     // Upload said tarball to the specified destination
+    // However, do not upload if performing a dry run
     try!(config.shell().status("Uploading", pkg.package_id().to_string()));
-    try!(transmit(&pkg, &tarball, &mut registry));
+    if !dry {
```

Comment:
Could this check be pushed into the `transmit` function actually? There's a few more errors that can happen right before the publish and it'd be nice to catch those.